### PR TITLE
ADEN-10794 Adding Stroer integration

### PR DESCRIPTION
**Pull request by Mariusz8Walczak**

### CODE REVIEW

File: src/ad-services/stroer/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+			utils.logger(logGroup, 'disabled');
+
+			return Promise.resolve();
+		}
+
+		const libraryUrl = `//js.adscale.de/map.js`;
+
+		utils.logger(logGroup, 'loading', libraryUrl);
+
+		return utils.scriptLoader.loadScript(libraryUrl, 'text/javascript', true).then(() => {
```
Comment (preludek) [inline src/ad-services/stroer/index.ts:17]: You don't need to pass `'text/javascript'` and `true` here. Those are default values in scriptLoader.loadScript method.

File: src/ad-services/stroer/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+			utils.logger(logGroup, 'ready');
+		});
+	}
+}
+
+export const stroer = new Stroer();
```
Comment (preludek) [inline src/ad-services/stroer/index.ts:23]: This class is not used anywhere. You need to call it in on platforms in callExternals, fe:
https://github.com/Wikia/ad-engine/blob/dev/platforms/ucp-desktop/modes/ucp-desktop-ads.mode.ts#L69

**Review by preludek — CHANGES_REQUESTED** [review]

**Review by Loczi94 — DISMISSED** [review]
Label major/minor/patch is missing.

What is more, unit tests are missing. We have unit tests for all services in `ad-engine/spec/ad-services.`

File: src/ad-services/stroer/index.ts (hunk 1 above)
Comment (Mariusz8Walczak) [reply src/ad-services/stroer/index.ts:17]: thanks

File: src/ad-services/stroer/index.ts (hunk 2 above)
Comment (Mariusz8Walczak) [reply src/ad-services/stroer/index.ts:23]: thanks, done

**Comment by Mariusz8Walczak** [comment]
There are no test, becouse when ic flag is on, script will be loaded, now is false, I think makeing test that are depended on something external for very basic code, that was used many times is mistake.

**Comment by Loczi94** [comment]
In tests we don't read icVar from Instant Config service. We set it to a specific value in ad-engine context. I know it is a dummy test but better have one dummy test than none.

File: platforms/shared/setup/base-context.setup.ts [hunk 3]
```diff
@@ -133,6 +133,7 @@ export class BaseContextSetup implements DiProcess {
 			'services.realVu.enabled',
 			this.instantConfig.get('icRealVu') && context.get('wiki.opts.enableRealVu'),
 		);
+		context.set('services.stroer.enabled', this.instantConfig.get('icStroer'));
```
Comment (dmnsobczak) [inline platforms/shared/setup/base-context.setup.ts:136]: Can we move it to L144 to keep alphabetical order?

File: platforms/ucp-mobile/modes/ucp-mobile-ads.mode.ts [hunk 4]
```diff
@@ -51,6 +52,7 @@ export class UcpMobileAdsMode implements DiProcess {
 		iasPublisherOptimization.call();
 		confiant.call();
 		realVu.call();
+		stroer.call();
```
Comment (dmnsobczak) [inline platforms/ucp-mobile/modes/ucp-mobile-ads.mode.ts:55]: Missing this set for `platforms/ucp/modes/ucp-ads-mode.ts`

File: src/ad-services/stroer/index.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { context, utils } from '@ad-engine/core';
+
+const logGroup = 'Stroer';
```
Comment (dmnsobczak) [inline src/ad-services/stroer/index.ts:3]: Let's keep (sadly, not always followed :<) conversion to use lowercase + dash for log groups :)

**Review by dmnsobczak — CHANGES_REQUESTED** [review]
Missing UCP (Oasis) implementation.

File: src/ad-services/stroer/index.ts (hunk 5 above)
Comment (Mariusz8Walczak) [reply src/ad-services/stroer/index.ts:3]: ok

File: platforms/ucp-mobile/modes/ucp-mobile-ads.mode.ts (hunk 4 above)
Comment (Mariusz8Walczak) [reply platforms/ucp-mobile/modes/ucp-mobile-ads.mode.ts:55]: Ok, I will add

File: src/ad-services/stroer/index.ts (hunk 5 above)
Comment (Mariusz8Walczak) [reply src/ad-services/stroer/index.ts:3]: added lowercase

File: platforms/shared/setup/base-context.setup.ts (hunk 3 above)
Comment (Mariusz8Walczak) [reply platforms/shared/setup/base-context.setup.ts]: L143 is done

**Review by dmnsobczak — APPROVED** [review]

**Review by preludek — APPROVED** [review]

File: src/ad-services/index.ts [hunk 6]
```diff
@@ -10,5 +10,6 @@ export * from './instant-config';
 export * from './nielsen';
 export * from './permutive';
 export * from './real-vu';
+export * from './stroer';
```
Comment (dmnsobczak) [inline src/ad-services/index.ts:13]: `ST` should be after `SI` :D

**Pull request merged**